File: backend/java/localpos-user-service/src/main/java/com/pds/localpos/userservice/repository/RoleRepository.java
package com.pds.localpos.userservice.repository;

import com.pds.localpos.userservice.model.Role;
import com.pds.localpos.userservice.model.RoleName;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Set;

@Repository
public interface RoleRepository extends JpaRepository<Role, String> {

    Set<Role> findByNameIn(Set<RoleName> names);
}

File: backend/kotlin/localpos-user-service-kt/src/main/kotlin/com/pds/localpos/userservice/repository/RoleRepository.kt
package com.pds.localpos.userservice.repository

import com.pds.localpos.userservice.model.Role
import com.pds.localpos.userservice.model.RoleName
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface RoleRepository : JpaRepository<Role, String> {

    fun findByNameIn(names: Collection<RoleName>): Set<Role>
}
